widget/widget_date_input.go

```diff
@@ -0,0 +1,208 @@
+package widget
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/container"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+	"time"
+)
+
+var _ fyne.Focusable = (*JDateInputWidget)(nil)
+var _ fyne.Tappable = (*JDateInputWidget)(nil)
+
+// Custom Date Input Widget using RichText
+type JDateInputWidget struct {
```

Comment:
Someone using this will get `widget.JDateInputWidget` which stutters. No need to say "widget" again. Also, what is the "J" for?

The README mentions `DateEntry` so why not call it that?

Reply:
I want to prefix my widget names with J no special meaning

Reply:
We need our APIs to be consistent and be easy to understand. I would suggest `DateInput`.